wiki/achievements.py

```diff
@@ -10,20 +13,25 @@ def process_file(f):
     with open(f) as file:
         for line in file:
             l = line.strip().split('\t')
-            if len(l) != 2:
+            if len(l) != 3:
                 continue
-            (k, v) = l
-            names[k] = v
+            (k, v, y) = l
+            names[k] = (v, y)
     return names
 
 
-REGEX_IN_DATE = r".*in\s*(?:[^ ,]*?)?\s*\d\d\d\d.*"
+REGEX_IN_DATE = r".*in\s*(?:[^ ,]*?)?\s*(\d\d\d\d).*"
 
 
-def process_page(id):
+def process_page(id, birth_year):
     page = wikipedia.page(pageid=id)
     in_date_regex = re.compile(REGEX_IN_DATE, re.IGNORECASE)
     tokenizer = nltk.data.load('tokenizers/punkt/english.pickle')
+
+    text_to_date = lambda x, y: datetime.strptime(x, y).date()
+
+    birth_date = text_to_date(birth_year, '%Y-%m-%d+%H:%M')
```

Comment:
@pepol yeah, see for instance [this URL](http://dbpedia.org/snorql/?query=PREFIX+dbo%3A+%3Chttp%3A%2F%2Fdbpedia.org%2Fontology%2F%3E%0D%0A%0D%0ASELECT+%3Fn%2C+%3Fs%2C+%3Fb+WHERE++{%0D%0A++++++++++++%3Fp+a+%3Chttp%3A%2F%2Fdbpedia.org%2Fontology%2FPerson%3E+.%0D%0A++++++++++++%3Fp+dbpedia-owl%3AwikiPageID+%3Fs+.%0D%0A++++++++++++%3Fp+dbpedia-owl%3AbirthDate+%3Fb+.%0D%0A++++++++++++%3Fp+foaf%3Aname+%3Fn+.%0D%0A++++++++}+LIMIT+10)